### PR TITLE
Fallback mechanism for supporting Pandas API calls

### PR DESCRIPTION
**Pull request by rohankumar42**
Initial mechanism which allows Pandas-like API calls to fallback to their Pandas counterparts, in cases where we have not yet implemented corresponding SQLite operations. Note that this is just a proof of concept, and only a couple of Pandas operations are supported via fallbacks with this change. Will need much more engineering work to include and test the rest.

This is an initial step towards #6.

### CODE REVIEW

File: pandasql/core.py [hunk 1]
```diff
@@ -175,6 +176,12 @@ def all(self):
     def __getattr__(self, attr):
         if attr in self.columns:
```
Comment (arjunrawal4) [inline pandasql/core.py]: Just a thing to remember - we will have weird/unexpected bugs if users name columns with function names

File: pandasql/cost_model.py [hunk 2]
```diff
@@ -37,29 +39,42 @@ def _out_of_memory(df, graph):
...
-    return isinstance(df, ps.core.Limit)
+    if isinstance(df, ps.core.Limit):
+        return True
 
 
 def _deep_dependency_graph(df, graph):
     ancestors_by_depth = _get_ancestors_by_depth(df, graph)
     depth = len(ancestors_by_depth)
-    return depth > 5
+    if depth > 5:
```
Comment (arjunrawal4) [inline pandasql/cost_model.py:64]: Maybe add a TODO to move all the numbers into some sort of configuration object?

File: pandasql/cost_model.py [hunk 3]
```diff
@@ -37,29 +39,42 @@ def _out_of_memory(df, graph):
...
+        return True
+
+
+def _fallback_operation(df, graph):
+    # Rule does not apply if the result of fallback is already cached
+    fallbacks = _filter_ancestors(df, graph, lambda x: x.result is None and
```
Comment (arjunrawal4) [inline pandasql/cost_model.py:75]: In a case where the last operation (say a df.describe) cannot be done in current sqlite translation we always run in pandas right now. Could/should we rerun the offload engine on every recursive compute() call to allow offloading for most of a query?

File: pandasql/api_status.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    'to_json',
+    'to_numpy',
+    'to_pickle',
+    'update'
+}
+
+SUPPORTED_VIA_FALLBACK = {
+    'nlargest',
+    'nsmallest',
+    # TODO: try to add all unsupported pandas ops
```
Comment (arjunrawal4) [inline pandasql/api_status.py:27]: Is the hold on this just adding tests?

File: pandasql/core.py (hunk 1 above)
Comment (rohankumar42) [reply pandasql/core.py]: Yes, good point. Pandas has the same problem. In such a case in Pandas, the method takes precedence over the column. I'll change this to reflect that.

File: pandasql/cost_model.py (hunk 2 above)
Comment (rohankumar42) [reply pandasql/cost_model.py:64]: Will do.

File: pandasql/cost_model.py (hunk 3 above)
Comment (rohankumar42) [reply pandasql/cost_model.py:75]: Oh interesting. We should look into this when we look into offloading partial queries. I'll put a TODO.

File: pandasql/api_status.py (hunk 4 above)
Comment (rohankumar42) [reply pandasql/api_status.py:27]: Not just on adding tests. A lot of the API is not simple `DataFrame -> DataFrame` function calls that we can fallback to like this. We'd need to figure out a way to support those too.

**Review by arjunrawal4 — APPROVED** [review]

**Pull request merged**